httpsredirect.go

```diff
@@ -1,10 +1,27 @@
 package gokrazy
 
-import "net/http"
+import (
+	"net/http"
+	"strings"
+)
 
 // TODO: Configurable https-port
-func httpsRedirect(w http.ResponseWriter, r *http.Request) {
-	r.URL.Host = r.Host
-	r.URL.Scheme = "https"
-	http.Redirect(w, r, r.URL.String(), http.StatusFound) // Redirect to https
+func httpsRedirect(handler http.Handler) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// get hostname without port
+		target := r.URL.Host
```

Comment:
Unfortunately, this approach isn’t safe: the “Host” HTTP header is under control by the attacker, not the server.

We need to look at https://golang.org/pkg/net/http/#Request RemoteAddr field. See the existing `IsInPrivateNet` function for how to compare IP addresses with networks, and then compare with just 127.0.0.0/8 and ::1/128.